packages/x-tree-view/src/useTreeItem2/useTreeItem2.types.ts

```diff
@@ -81,11 +82,23 @@ export type UseTreeItemIconContainerSlotProps<ExternalProps = {}> = ExternalProp
 
 export interface UseTreeItem2LabelSlotOwnProps {
   children: React.ReactNode;
+  onDoubleClick: MuiCancellableEventHandler<React.MouseEvent>;
+  editable?: boolean;
 }
 
 export type UseTreeItem2LabelSlotProps<ExternalProps = {}> = ExternalProps &
   UseTreeItem2LabelSlotOwnProps;
 
+export type UseTreeItem2LabelInputSlotOwnProps = {
+  onBlur: MuiCancellableEventHandler<React.FocusEvent<HTMLInputElement>>;
+  value: React.ReactNode;
+  onKeyDown: MuiCancellableEventHandler<React.KeyboardEvent<HTMLInputElement>>;
+  onChange: React.ChangeEventHandler<HTMLInputElement>;
```

Comment:
You are missing `data-element` here.
And I would add some JSDoc to it so that we remember why it exists.
I would love to find a way to drop it in the future, but I don't think it's worth postponing the feature to find a solution :+1: